src/python/ensembl/io/genomio/integrity.py

```diff
@@ -21,211 +21,147 @@
+        if name in self.lengths:
+            return self.lengths[name]
+        raise KeyError(f"No length available for key {name}")
 
-    def get_manifest(self) -> Dict[str, Any]:
+    def _add_error(self, error: str) -> None:
+        self.errors.append(error)
+
+    def get_manifest(self, manifest_path: PathLike) -> Dict[str, Any]:
         """Load the content of a manifest file.
 
         Returns:
             Dict: Content of the manifest file.
         """
-        with open(self.manifest_file) as manifest_fh:
+        manifest_path = Path(manifest_path)
+        with manifest_path.open("r") as manifest_fh:
             manifest = json.load(manifest_fh)
 
             # Use dir name from the manifest
             for name in manifest:
                 if "file" in manifest[name]:
-                    file_name = manifest[name]["file"]
-                    file_name = path.join(path.dirname(self.manifest_file), file_name)
+                    file_path = manifest_path.parent / manifest[name]["file"]
                     # check if the md5sum is correct
                     md5sum = manifest[name]["md5sum"]
-                    self.check_md5sum(file_name, md5sum)
+                    self._check_md5sum(file_path, md5sum)
 
-                    manifest[name] = file_name
+                    manifest[name] = file_path
                 else:
                     for f in manifest[name]:
                         if "file" in manifest[name][f]:
-                            file_name = manifest[name][f]["file"]
-                            file_name = path.join(path.dirname(self.manifest_file), file_name)
+                            file_path = manifest_path.parent / manifest[name][f]["file"]
                             # check if the md5sum is correct
                             md5sum = manifest[name][f]["md5sum"]
-                            self.check_md5sum(file_name, md5sum)
+                            self._check_md5sum(file_path, md5sum)
 
-                            manifest[name][f] = file_name
+                            manifest[name][f] = file_path
             return manifest
 
-    def check_integrity(self):
-        """Load files listed in the manifest.json and check the integrity.
-        Check if the files are correct by verifying the MD5 hash.
-        Check if translation, functional annotation and sequence region ids
-        and lengths are consistent with the information in gff.
-        Compare sequence length from fasta_dna file to seq_region.json metadata.
+    def _check_md5sum(self, file_path: Path, md5sum: str) -> None:
+        """Verify the integrity of the files in manifest.json.
+
+            An MD5 hash is generated using the path provided which is then compared to the hash
+            in manifest.json.
+            Errors are stored in self.errors
+
+        Args:
+            Path: The path for each file in the genome.
```

Comment:
```suggestion
            file_path: Path to a genome file.
```
When I have read "each" I have thought `file_path` was a list, but it is a single path.